src/banded/BandedMatrix.jl

```diff
@@ -1007,3 +1007,57 @@ function resize(A::BandedSubBandedMatrix, n::Integer, m::Integer)
     l,u = bandwidths(A)
     _BandedMatrix(reshape(resize!(vec(copy(bandeddata(A))), (l+u+1)*m), l+u+1, m), n, l,u)
 end
+
+function Base.sum(A::BandedMatrix)
+    l,u = bandwidths(A)
+    #For empty matrices
+    if(l + u < 0)
+        return convert(eltype(A),0)
+    end
+    height,width = size(A)
+    #Only get nonempty bands
+    lower, upper = min(height-1,l), min(width-1,u)
+    data = zeros(1+lower+upper,min(height,width))
```

Comment:
This doesn't have the right type. Also allocating memory is a bad idea. Here we just want to do something like
```julia
ret = zero(eltype(A))
for j = 1:n, k = ...
   ret += ...
end
```